src/persist-client/src/read.rs

```diff
@@ -525,6 +531,12 @@ where
         &self,
         as_of: Antichain<T>,
     ) -> Result<Vec<ReaderEnrichedHollowBatch<T>>, Since<T>> {
+        info!(
```

Comment:
I'm worried that several of these are going to be too spammy to merge to main. I was imagining something like just the `Force expiring reader`/`Force expiring writer` log plus including the shard and reader_id/writer_id in the panics in State::reader and State::writer. (now that I see it, I also like your addition of shard to the missing blob panic)

Reply:
Ooh adding the reader/writer into the State panics is nice. Updated to reduce the logginess